File: EPI/7.9/swift/7.9_test.swift
import Darwin

let tests: [(roman: String, int: Int)] = [
    ("", 0),
    ("I", 1),
    ("III", 3),
    ("IV", 4),
    ("VI", 6),
    ("XXXXXIIIIIIIII", 59),
    ("LVIIII", 59),
    ("LIX", 59),
    ("MDC", 1600),
    ("CMCD", 1300),
]

for test in tests {
    var actual = toInt(roman: test.roman)
    guard actual == test.int else {
        print("For roman '\(test.roman)', expected int to be \(test.int), but was \(actual)")
        exit(1)
    }
}

File: EPI/7.9/typescript/7.9_test.ts
import evaluate from './7.9';

declare function require(name: string): any;
const assert = require('assert');

const tests: {roman: string, value: number}[] = [
    {roman: '', value: 0},
    {roman: 'I', value: 1},
    {roman: 'III', value: 3},
    {roman: 'IV', value: 4},
    {roman: 'VI', value: 6},
    {roman: 'XXXXXIIIIIIIII', value: 59},
    {roman: 'LVIIII', value: 59},
    {roman: 'LIX', value: 59},
    {roman: 'MDC', value: 1600},
    {roman: 'CMCD', value: 1300},
];

for (const test of tests) {
    const actual = evaluate(test.roman);
    assert.strictEqual(actual, test.value, `For ${test.roman}, ` +
        `expected value to be ${test.value}, but was ${actual}`);
}
